### PR TITLE
Adds withFeatureToggles

### LINKED ISSUE #116: Implement v2 withFeatureToggles
part of #103

### PR DESCRIPTION
**Pull request by jwicks31**
Resolves #116

### CODE REVIEW

**Comment by kennylavender** [comment]
And add integration tests

File: src/test/with-feature-toggles.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import ReactDOMServer from 'react-dom/server';
+import { describe } from 'riteway';
+import dom from 'cheerio';
+import { withFeatureToggles } from '../with-feature-toggles';
+
+import { Consumer } from '../context';
+
+const render = ReactDOMServer.renderToStaticMarkup;
+
+const createTestHOCComponent = componentName => ({ propCheck } = {}) => (
```
Comment (kennylavender) [inline src/test/with-feature-toggles.js]: Is a dynamic component name necessary for testing withFeatureToggles? Think about what we are actually testing about withFeatureToggles.

**Comment by jwicks31** [comment]
Is the integration test with configureFeature all we need? Since that is basically just passing things down to Feature?

File: src/test/with-feature-toggles.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import { Consumer } from '../context';
+
+const render = ReactDOMServer.renderToStaticMarkup;
+
+const createTestComponent = () => ({ propCheck } = {}) => (
+  <Consumer propCheck={propCheck}>
+    {features => {
+      return (
+        <div>
+          <div className="component">{features.toString()}</div>
```
Comment (kennylavender) [inline src/test/with-feature-toggles.js]: cool, we can probably rename the class name to something that makes a bit more sense now, like maybe features

File: src/test/with-feature-toggles.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import { describe } from 'riteway';
+import dom from 'cheerio';
+import { withFeatureToggles } from '../with-feature-toggles';
+
+import { Consumer } from '../context';
+
+const render = ReactDOMServer.renderToStaticMarkup;
+
+const createTestComponent = () => ({ propCheck } = {}) => (
+  <Consumer propCheck={propCheck}>
```
Comment (kennylavender) [inline src/test/with-feature-toggles.js]: We don't need to pass any props to a `Consumer` component, it does nothing with them.

**Comment by kennylavender** [comment]
Yup, that's fine, I probably would have just used the `Feature` component directly, but either way works.

File: src/test/with-feature-toggles.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  {
+    const { assert } = should();
+    const Component = createTestComponent();
+    const features = [];
+    const Page = withFeatureToggles({ features })(Component);
+
+    const $ = dom.load(render(<Page />));
+
+    assert({
```
Comment (kennylavender) [inline src/test/with-feature-toggles.js]: we don't need to test that the component exists, the `.text()` tests tell us the component exists, so we can remove all the `.length` tests.

**Pull request merged**